src/bp/ui-studio/src/web/views/OneFlow/diagram/nodes/Block/index.tsx

```diff
@@ -55,11 +58,22 @@ const BlockWidget: FC<Props> = ({
   addCondition,
   getCurrentLang
 }) => {
-  const [expanded, setExpanded] = useState(node.isNew)
+  let expandedNodes = JSON.parse(storage.get(EXPANDED_NODES_KEY) || '[]')
```

Comment:
When you get content from the storage, the parsing must always be done in a try/catch. If the array gets corrupted or some text is missing, the studio will always crash on that bot. 

Another point, this implementation means that if you have 40 nodes on the studio, they will have to parse that key 40 times and each one will have their own "implementation". The management of the expanded state must be central (done by the diagram), and nodes simply "bubble up" their expanded/collapsed state